### ISSUE TITLE
How do I fix domain sizes in a stencil?

### ISSUE THREAD

**Issue by tbennun** [GridTools/gt4py#990]
When generating stencils, it would be beneficial to have a mode where all the sizes (and strides) are fixed in the generated code. It should especially help in the CPU case on stencils with many fields, which easily start spilling registers.

Thanks in advance!

**Comment by tbennun** [comment]
When I asked this, I wasn't sure whether `freeze` also freezes the dimensions. I think it would be a good feature to be able to configure that requirement, rather than let the backend decide without involving the user. Thoughts?

**Comment by jdahm** [comment]
> I think it would be a good feature to be able to configure that requirement, rather than let the backend decide without involving the user

I'm not sure I understand the idea. The current way it works is that the user has to invoke the `freeze` method and pass it the domain and origins. How would you envision it working?

**Comment by tbennun** [comment]
How I would envision it is that `freeze` would accept another optional parameter (e.g., `domain_size=None`). If it is not given, only the current elements would be frozen. If it is, the size would be frozen as well, which could be used by the backends to shape-infer the sizes and strides of the domain and fields.

**Comment by jdahm** [comment]
Hi @tbennun, thanks for your message. I'm still a bit confused: What I'm gathering from your message is that you're asking about inferring sizes and strides from the overall compute domain.

What I thought you were going to get at (and is a nice idea for optimization) is to add a way to "freeze" not only the origin of each field and the compute `domain` size, but also the `shape` of each input ndarray. Is that what you're getting at? If not, let's VC tomorrow or soon to understand this better.

**Comment by tbennun** [comment]
You are correct. What I (mistakenly) thought was that all array shapes can be inferred from the domain size and their use in the stencils.
